dll/win32/browseui/regtreeoptions.cpp

```diff
@@ -1,59 +1,565 @@
 /*
- * ReactOS Explorer
- *
- * Copyright 2009 Andrew Hill <ash77 at domain reactos.org>
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation; either
- * version 2.1 of the License, or (at your option) any later version.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ * PROJECT:     browseui
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Common registry based settings editor
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
  */
 
 #include "precomp.h"
 
-CRegTreeOptions::CRegTreeOptions()
+#define HKEY_GRAYED HKEY_REGTREEOPTION_GRAYED
+#define MAXVALUEDATA 100 // Max size for CheckedValue/UncheckedValue/DefaultValue
+
+enum {
+    TYPE_CHECK = 0,
+    TYPE_RADIO,
+    TYPE_GROUP,
+    TYPE_INVALID,
+    STATEIMAGESPERTYPE = 4,
+    STATE_CHECKOFF = TYPE_CHECK * STATEIMAGESPERTYPE,
+    STATE_CHECKON,
+    STATE_RADIOOFF = TYPE_RADIO * STATEIMAGESPERTYPE,
+    STATE_RADIOON,
+    STATE_GROUP    = TYPE_GROUP * STATEIMAGESPERTYPE,
+};
+
+C_ASSERT((STATE_CHECKON & 1) && (STATE_RADIOON & 1) && STATE_RADIOON < STATE_GROUP);
+
+static HBITMAP CreatDibBitmap(HDC hDC, UINT cx, UINT cy, UINT bpp)
 {
+    BITMAPINFO bi;
+    LPVOID pvBits;
+
+    ZeroMemory(&bi, sizeof(bi));
+    bi.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+    bi.bmiHeader.biWidth = cx;
+    bi.bmiHeader.biHeight = cy;
+    bi.bmiHeader.biPlanes = 1;
+    bi.bmiHeader.biBitCount = bpp;
+    bi.bmiHeader.biCompression = BI_RGB;
+    return CreateDIBSection(hDC, &bi, DIB_RGB_COLORS, &pvBits, NULL, 0);
 }
 
-CRegTreeOptions::~CRegTreeOptions()
+static HIMAGELIST CreateStateImages(UINT ImageSize)
 {
+    enum { bpp = 32 };
+    UINT Failed = FALSE, State;
+    HIMAGELIST hIL = ImageList_Create(ImageSize, ImageSize, ILC_MASK | bpp, 4, 0);
```

Comment:
4 images or 8? (I see 8 ImageList_Add calls below).
If the Create call succeeds for the number of images you initially specify, and if you add as many images with the Add calls as initially specified, I wonder whether it's possible the Add calls could fail or not...

Reply:
It is just a hint for the initial size, you can add more later.